### PR TITLE
Add InboxPopover and extract InboxContainer

### LINKED ISSUE #1654: Add inbox popover
### Story

As a _worker assigned to a few tasks_, I want to _see who has mentioned me in a comment_, so that _I can easily access the right task without switching the context_. 

### Description

1. Clicking the envelope icon in the top right area opens up a popover with a list of notifications.
2. The blue vertical stripe shows unread notifications (if this is super complex, it can be a taco, or can be shipped later, but highly recommended)
3. Ignore the pink stripe for now – they are for time-dependent notifications (like a due date, or time required to do something). 
4. Notice the pink dot – it shows up if new notifications appear. It disappears after opening the popover regardless if someone opened the notification or not. We do not show any numbers there. Opening the popover means that notifications are read. 
5. When the popover is opened, the envelope icon has a blue colour.
6. Clicking "see all" opens a new page with a fullscreen view of the notifications. 

### Design 

[Figma link](https://www.figma.com/file/OD53CoZpRKVqDdKsMYGldko8/Inbox?node-id=194%3A0)

image

### This issue is complete when...

- [x] The inbox icon has a red indicator when there are unread inbox items
- [x] The inbox icon opens a popover with the inbox content
- [x] The inbox icon is active when the popover is open
- [x] The notifications are marked as read when the popover is opened
- [x] The inbox content is adjusted to fit in the popover according to the design (if necessary)
- [x] A 'see all' button is added to the bottom of the inbox content, which links to the full-screen view
- [x] 🌮The content of the popover is visibly limited to 4 inbox items (with vertical overflow scroll)

### Subtasks

- [x] Add unread indicator to icon
- [x] Define popover
- [x] Trigger popover with icon and mark icon as active
- [x] Adjust popover style if necessary
- [x] Add 'see all' button
- [x] 🌮Limit the content to 4 visible items and provide vertical scroll for overflow

### Suggestions

* From @rdig : we need to bear in mind that colony/user addresses (when a label isn't available) will need to be shortened. This might cause visual issues.
* Also from @rdig : Look at the network health/gas station popovers. It might be possible to extract these components into something more generalised and use them for this.

It is more about checking the style of the inbox popover on figma.

 ~Fullscreen view is a taco or can be part of a different task (iceboxed for the future).~ The fullscreen view is already implemented AFAICT _—James_

The "X" icon to remove the notification from the list is a taco (see description on figma).

### PR DESCRIPTION
**Pull request by karolinkas**
## Description

Besides having a fullscreen view of the inbox we would like to have an inbox popover that can be expanded into the fullscreen view. It should have similar functionalities like the full version but only show 4 elements at a time. 

**Changes** 🏗
Extracted table of inbox elements from InboxContent component into a new component called `InboxContainer`, to be able to reuse the same table in the popover. That component now receives a `full` prop to decide about if it appears in a popover or as a fullscreen inbox.

## Done

- [x] Add unread indicator to icon
- [x] Define popover
- [x] Trigger popover with icon and mark icon as active
- [x] Adjust popover style if necessary
- [x] Add 'see all' button
- [x] 🌮Limit the content to 4 visible items and provide vertical scroll for overflow
Resolves #1654

### CODE REVIEW

**Comment by rdig** [comment]
@karolinkas what's the state here, is this awaiting review, has it been reviewed yet ?

**Comment by karolinkas** [comment]
@rdig It is waiting, there are no comments...

**Comment by rdig** [comment]
@JamesLefrere can you take some time to review it, please ?

**Comment by JamesLefrere** [comment]
Sorry! On it.

File: src/modules/users/components/Inbox/InboxIcon/InboxIcon.tsx [hunk 1]
```diff
@@ -23,18 +18,8 @@ interface Props {
...
-  hasUnreadActivities,
-}: Props) => (
-  <NavLink
-    to={INBOX_ROUTE}
-    className={hasUnreadActivities ? styles.inboxIconWCircle : styles.inboxIcon}
-    activeClassName={activeClassName}
-  >
-    <Icon name="envelope" title={title} />
-  </NavLink>
+const Inbox = ({ title = MSG.fallbackTitle }: Props) => (
```
Comment (JamesLefrere) [inline src/modules/users/components/Inbox/InboxIcon/InboxIcon.tsx]: The variable name here should probably be `InboxIcon`; the display name is correct, though. We can also wait for this to be fixed in #1775

File: src/modules/users/components/Inbox/InboxPopover/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/* @flow */
```
Comment (JamesLefrere) [inline src/modules/users/components/Inbox/InboxPopover/index.ts]: This comment can go (I guess it was from the generated file?)

File: src/modules/users/components/Inbox/InboxFullscreen/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/* @flow */
```
Comment (JamesLefrere) [inline src/modules/users/components/Inbox/InboxFullscreen/index.ts]: This comment can go (I guess it was from the generated file?)

File: src/modules/users/components/Inbox/InboxContainer/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+/* @flow */
```
Comment (JamesLefrere) [inline src/modules/users/components/Inbox/InboxContainer/index.ts]: This comment can go (I guess it was from the generated file?)

File: src/modules/users/components/Inbox/InboxContainer/InboxContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
+/* @flow */
```
Comment (JamesLefrere) [inline src/modules/users/components/Inbox/InboxContainer/InboxContainer.tsx]: This comment can go (I guess it was from the generated file?)

**Pull request merged**